src/manifolds/SymmetricPositiveSemidefiniteFixedRank.jl

```diff
@@ -0,0 +1,280 @@
+       AbstractEmbeddedManifold{𝔽,DefaultIsometricEmbeddingType} end
+
+function SymmetricPositiveSemidefiniteFixedRank(n::Int, k::Int, field::AbstractNumbers = ℝ)
+    return SymmetricPositiveSemidefiniteFixedRank{n,k,field}()
+end
+
+@doc raw"""
+    check_manifold_point(M::SymmetricPositiveSemidefiniteFixedRank{n,𝔽}, q; kwargs...)
+
+Check whether `q` is a valid manifold point on the [`SymmetricPositiveSemidefiniteFixedRank`](@ref) `M`, i.e.
+whether `p=q*q'` is a symmetric matrix of size `(n,n)` with values from the corresponding
+[`AbstractNumbers`](@ref) `𝔽`.
+The symmetry of `p` is not explicitly checked since by using `q` p is symmetric by construction.
+The tolerance for the symmetry of `p` can and the rank of `q*q'` be set using `kwargs...`.
+"""
+function check_manifold_point(
+    M::SymmetricPositiveSemidefiniteFixedRank{n,k,𝔽},
+    q;
+    kwargs...,
+) where {n,k,𝔽}
+    mpv =
+        invoke(check_manifold_point, Tuple{supertype(typeof(M)),typeof(q)}, M, q; kwargs...)
+    mpv === nothing || return mpv
+    p = q * q'
+    if rank(p * p'; kwargs...) < k
+        return DomainError(
+            rank(p * p'; kwargs...),
+            "The point $(p) does not lie on $M, since its rank is less than $(k).",
+        )
+    end
+    return nothing
+end
+
+"""
+    check_tangent_vector(M::SymmetricPositiveSemidefiniteFixedRank{n,k,𝔽}, p, X; check_base_point = true, kwargs... )
+
+Check whether `X` is a tangent vector to manifold point `p` on the
+[`SymmetricPositiveSemidefiniteFixedRank`](@ref) `M`, i.e. `X` has to be a symmetric matrix of size `(n,n)`
+and its values have to be from the correct [`AbstractNumbers`](@ref).
+The optional parameter `check_base_point` indicates, whether to call
+ [`check_manifold_point`](@ref)  for `p`.
+The tolerance for the symmetry of `p` and `X` can be set using `kwargs...`.
+"""
+function check_tangent_vector(
+    M::SymmetricPositiveSemidefiniteFixedRank{n,k,𝔽},
+    q,
+    Y;
+    check_base_point = true,
+    kwargs...,
+) where {n,k,𝔽}
+    if check_base_point
+        mpe = check_manifold_point(M, q; kwargs...)
+        mpe === nothing || return mpe
+    end
+    mpv = invoke(
+        check_tangent_vector,
+        Tuple{supertype(typeof(M)),typeof(q),typeof(Y)},
+        M,
+        q,
+        Y;
+        check_base_point = false, # already checked above
+        kwargs...,
+    )
+    return mpv
+end
+
+function decorated_manifold(::SymmetricPositiveSemidefiniteFixedRank{N,K,𝔽}) where {N,K,𝔽}
+    return Euclidean(N, K; field = 𝔽)
+end
+
+@doc raw"""
+    distance(M::SymmetricPositiveSemidefiniteFixedRank, p, q)
+
+Compute the distance between two points `p`, `q` on the
+[`SymmetricPositiveSemidefiniteFixedRank`](@ref), which is the Frobenius norm of $Y$ which
+minimizes $\lVert p - qY\rVert$ with respect to $Y$.
+"""
+distance(M::SymmetricPositiveSemidefiniteFixedRank, p, q) = norm(M, p, log(M, p, q))
+
+embed!(::SymmetricPositiveSemidefiniteFixedRank, q, p) = copyto!(q, p)
```

Comment:
Shouldn't embed perform the decomposition of `p` (n by n matrix) and write only the (n by k) part to `q`?

Reply:
No. This is currently modeled as embedding this manifold into `n \times k` matrices to “steal” their metric, so it is not the classical embedding one would first think of ( doing qq^T, but maybe the p here is misleading). So it is an embedding of the parametrizarion (q), wince (if k is far less then n) you actually never want to store the large `n \times n` matrix. So that's also one point where I noticed, since embedding might be misleading, naming it quotient would fit better.

Reply:
OK, I see.